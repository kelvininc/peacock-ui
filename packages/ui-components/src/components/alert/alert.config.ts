import { EIconName } from '../icon/icon.types';
import { EAlertType } from './alert.types';

export const ALERT_ICON_NAMES: { [key in EAlertType]: EIconName } = {
	[EAlertType.Info]: EIconName.Error,
	[EAlertType.Error]: EIconName.Error,
	[EAlertType.Success]: EIconName.Success,
	[EAlertType.Warning]: EIconName.Warning
};
