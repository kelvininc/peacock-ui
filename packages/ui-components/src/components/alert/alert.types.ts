import { EComponentSize } from '../../types';

export enum EAlertType {
	Info = 'info',
	Error = 'error',
	Success = 'success',
	Warning = 'warning'
}

export interface IAlertConfig {
	/** (required) Defines the type of alert to show, possible values are defined in `EAlertType` */
	type: EAlertType;
	/** (optional) Defines the size of the component, defaults to EComponentType.Large */
	size?: EComponentSize;
	/** (optional) Defines if the icon should be shown, defaults to `true` */
	showIcon?: boolean;
	/** (required) Defines the label (title) text to show */
	label: string;
	/** (optional) Defines the description text to show */
	description?: string;
}
