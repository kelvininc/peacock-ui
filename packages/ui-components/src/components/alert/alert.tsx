import { Component, Host, Prop, h } from '@stencil/core';
import { EAlertType, IAlertConfig } from './alert.types';
import { ALERT_ICON_NAMES } from './alert.config';
import { EComponentSize } from '../../types';

@Component({
	tag: 'kv-alert',
	styleUrl: 'alert.scss',
	shadow: true
})
export class KvAlert implements IAlertConfig {
	/** @inheritdoc */
	@Prop({ reflect: true }) type!: EAlertType;
	/** @inheritdoc */
	@Prop({ reflect: true }) size?: EComponentSize = EComponentSize.Large;
	/** @inheritdoc */
	@Prop({ reflect: true }) showIcon?: boolean = true;
	/** @inheritdoc */
	@Prop({ reflect: true }) label!: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) description?: string;

	render() {
		return (
			<Host>
				<div class="alert-container">
					<div
						class={{
							'alert': true,
							[`alert--type-${this.type}`]: true,
							'alert--size-small': this.size === EComponentSize.Small
						}}
					>
						<div class="information">
							{this.showIcon && <kv-icon name={ALERT_ICON_NAMES[this.type]} class="icon" />}
							<div class="text-container">
								<div class="title">{this.label}</div>
								<slot name="description">{this.description && <div class="description">{this.description}</div>}</slot>
							</div>
						</div>
						<slot name="actions"></slot>
					</div>
				</div>
			</Host>
		);
	}
}
