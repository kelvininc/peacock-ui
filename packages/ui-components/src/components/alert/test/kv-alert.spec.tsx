import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvAlert } from '../alert';

describe('Alert (unit tests', () => {
	let page: SpecPage;

	describe('when using required props and defaults', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvAlert],
				html: '<kv-alert type="info" label="Main Message" />'
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});

	describe('when using required props and description', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvAlert],
				html: '<kv-alert type="info" label="Main Message" description="Secondary Message" />'
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});

	describe('when using required props and showIcon `false`', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvAlert],
				html: '<kv-alert type="info" label="Main Message" show-icon="false" />'
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});
});
