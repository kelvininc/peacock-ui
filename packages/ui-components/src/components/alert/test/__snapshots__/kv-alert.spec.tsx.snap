// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert (unit tests when using required props and defaults should match the snapshot 1`] = `
<kv-alert label="Main Message" show-icon="" size="large" type="info">
  <mock:shadow-root>
    <div class="alert-container">
      <div class="alert alert--type-info">
        <div class="information">
          <kv-icon class="icon" name="kv-error"></kv-icon>
          <div class="text-container">
            <div class="title">
              Main Message
            </div>
          </div>
        </div>
        <slot name="actions"></slot>
      </div>
    </div>
  </mock:shadow-root>
</kv-alert>
`;

exports[`Alert (unit tests when using required props and description should match the snapshot 1`] = `
<kv-alert description="Secondary Message" label="Main Message" show-icon="" size="large" type="info">
  <mock:shadow-root>
    <div class="alert-container">
      <div class="alert alert--type-info">
        <div class="information">
          <kv-icon class="icon" name="kv-error"></kv-icon>
          <div class="text-container">
            <div class="title">
              Main Message
            </div>
            <div class="description">
              Secondary Message
            </div>
          </div>
        </div>
        <slot name="actions"></slot>
      </div>
    </div>
  </mock:shadow-root>
</kv-alert>
`;

exports[`Alert (unit tests when using required props and showIcon \`false\` should match the snapshot 1`] = `
<kv-alert label="Main Message" show-icon="false" size="large" type="info">
  <mock:shadow-root>
    <div class="alert-container">
      <div class="alert alert--type-info">
        <div class="information">
          <div class="text-container">
            <div class="title">
              Main Message
            </div>
          </div>
        </div>
        <slot name="actions"></slot>
      </div>
    </div>
  </mock:shadow-root>
</kv-alert>
`;
